content/docs/use-cases/ci-cd-for-machine-learning.md

```diff
@@ -0,0 +1,89 @@
+# Continuous Integration and Deployment for Machine Learning
+
+Applying DevOps methodologies to machine learning (MLOps) and data management
+(DataOps) is increasingly common. This means resource orchestration
+(provisioning servers), testing, and deployment to production (CI/CD --
+continuous integration and continuous delivery), as well as monitoring &
+feedback.
+
+![](https://static.iterative.ai/img/ml-pipeline.png) _Basic ML pipeline_
+
+The main benefits of CI/CD for ML are detailed below.
+
+### Automating Quality Assurance (QA)
+
+This concerns automating & enforcing testing. For example:
+
+- Enforcing data integrity with application-specific tests (such as validation
+  against a schema or verifying pipeline consistency).
+- Enforcing model integrity with application-specific tests (such as
+  input/output and performance validation).
+- When data or code is added/changed in a pull request (PR), automatically
+  running validation tests in the cloud.
+- Automatically generate metrics reports.
+
+### Continuous Machine Learning (CML)
+
+This concerns training in the cloud. For example:
+
+- Infrastructure orchestration: make CI systems provision and launch a GPU
+  instance, train a model, cleanly terminate, and pull results back.
+- Fine-tune on a schedule: set up jobs to pull in new data from regularly
+  updated external sources to retrain/refine deployed models.
+- Run a hyperparameter search: keep heavyweight resources and big data on CI
+  servers running overnight, and put your laptop to sleep.
+
+## Helpful Tools
+
+Normally, CI/CD is hard to set up, configure, and maintain -- especially for
+data and ML pipelines.
+
+![](https://static.iterative.ai/img/ml-vs-cicd.png) _Traditional ML meets CI/CD_
+
+DVC and CML can alleviate most (in not all) of the management headache
+([hidden technical debt](https://papers.nips.cc/paper/2015/file/86df7dcfd896fcaf2674f757a2463eba-Paper.pdf))
+in ML. You can automate all of the above without needing any additional
+configuration. Here are a few feature highlights:
+
+**Models, Data, and Metrics as Code**: DVC removes the need to manage databases,
```

Comment:
`DVC removes the need to manage databases` - it is a bit misleading statement since DVC is not a replacement for DBs.